membership.go

```diff
@@ -402,14 +431,24 @@ func multicastAnnounce(addr string) error {
 		logError(err)
 		return err
 	}
+	var fullLaddr string
+	if ipLen == net.IPv6len {
+		fullLaddr = fmt.Sprintf("[%s]:0", GetListenIP().String())
```

Comment:
- Is parsing strings our only option here? I fear that it may be, but I'm really not sure.
- Would you mind adding a comment above this `if` block? The string-parsing logic isn't obvious.

Reply:
I think its better to just create the struct by

```go
fullLaddr := &net.UDPAddr{
  IP: GetListenIP(),
  Port: 0,
}
```